edge-modules/TwinTester/Settings.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+
+    public class Settings
+    {
+        static readonly Lazy<Settings> DefaultSettings = new Lazy<Settings>(
+            () =>
+            {
+                IConfiguration configuration = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("config/settings.json", optional: true)
+                    .AddEnvironmentVariables()
+                    .Build();
+
+                return new Settings(
+                    configuration.GetValue<string>("IOTEDGE_DEVICEID", string.Empty),
+                    configuration.GetValue<string>("IOTEDGE_MODULEID", string.Empty),
+                    configuration.GetValue<int>("TwinUpdateSize", 1),
+                    configuration.GetValue<TimeSpan>("TwinUpdateFrequency", TimeSpan.FromSeconds(10)),
+                    configuration.GetValue<TimeSpan>("TwinUpdateFailureThreshold", TimeSpan.FromMinutes(1)),
+                    configuration.GetValue<TransportType>("TransportType", TransportType.Amqp_Tcp_Only),
+                    configuration.GetValue<Uri>("AnalyzerUrl", new Uri("http://analyzer:15000")),
+                    configuration.GetValue<string>("ServiceClientConnectionString"),
+                    configuration.GetValue<string>("StoragePath"),
+                    configuration.GetValue<bool>("StorageOptimizeForPerformance", true));
+            });
+
+        Settings(
+            string deviceId,
+            string moduleId,
+            int twinUpdateSize,
+            TimeSpan twinUpdateFrequency,
+            TimeSpan twinUpdateFailureThreshold,
+            TransportType transportType,
+            Uri analyzerUrl,
+            string serviceClientConnectionString,
+            string storagePath,
+            bool storageOptimizeForPerformance)
+        {
+            this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            this.ModuleId = Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId));
+            this.TwinUpdateSize = Preconditions.CheckRange(twinUpdateSize, 0);
+            this.TwinUpdateFrequency = Preconditions.CheckNotNull(twinUpdateFrequency);
+            this.TwinUpdateFailureThreshold = Preconditions.CheckNotNull(twinUpdateFailureThreshold);
+            this.TransportType = Preconditions.CheckNotNull(transportType);
+            this.AnalyzerUrl = Preconditions.CheckNotNull(analyzerUrl, nameof(analyzerUrl));
+            this.ServiceClientConnectionString = Preconditions.CheckNonWhiteSpace(serviceClientConnectionString, nameof(serviceClientConnectionString));
+            this.StoragePath = storagePath;
```

Comment:
No preconditions here as storage path can be empty string. I should check for null though.